chia-bls/src/signature.rs

```diff
@@ -30,7 +49,11 @@ impl Signature {
             blst_p2_from_affine(p2.as_mut_ptr(), &p2_affine.assume_init());
             p2.assume_init()
         };
-        let ret = Self(p2);
+        Ok(Self(p2))
+    }
+
+    pub fn from_bytes(buf: &[u8; 96]) -> Result<Self> {
```

Comment:
If this checks `is_valid`, what's the difference between the builtin check done by `p2_uncompress`?

Reply:
I don't know. I was not expecting `p2_uncompress()` would check if the point is on the curve, but it looks like it is. at least from the error code names.

Reply:
as far as I can tell, BLST doesn't have a version of `blst_p2_uncompress()` that doesn't also check if the point is on the curve. The only remaining check in `is_valid()` is the infinity check, which may also be redundant. I think this should be improved in a separate PR. I don't believe there's a correctness issue right now, just a performance issue